lib/vagrant-cloudstack/action/read_ssh_info.rb

```diff
@@ -70,8 +70,16 @@ def read_ssh_info(cloudstack, machine)
             end
           end
 
+          nic_ip_address = server.nics[0]['ipaddress']
+
+          unless domain_config.ssh_network_id.nil?
+            # When ssh_network_id is specified, use the IP address that is linked to the network_id of nic
+            ssh_nic = server.nics.find { |nic| nic["networkid"] == domain_config.ssh_network_id }
```

Comment:
As with PR #148 , probably support both IDs and Names?

Actually, could you name the parameter `ssh_network` and just check/search on both `"networkid"` and (if that didn't resolve) also on `"networkname"`?